src/utils/focusTrapComponent.ts

```diff
@@ -42,12 +42,15 @@ export function connectFocusTrap(component: FocusTrapComponent): void {
     return;
   }
 
-  if (focusTrapEl.tabIndex == null) {
-    focusTrapEl.tabIndex = -1;
+  if (focusTrapEl.tabIndex === null) {
+    focusTrapEl.setAttribute("tabindex", "-1");
   }
 
   const focusTrapOptions: FocusTrapOptions = {
-    clickOutsideDeactivates: true,
+    clickOutsideDeactivates: (event: MouseEvent | TouchEvent) => {
+      return !event.composedPath().find((el) => (el as HTMLElement) === focusTrapEl);
+    },
+    allowOutsideClick: true,
```

Comment:
Nitpick: can you sort the props alphabetically? Makes it easier to know where they go and to look up.